### DISCUSSION TITLE
Residue-specific proteomics (isoTOP-ABPP) with DIA-NN

### DISCUSSION THREAD

**Discussion by StephanHacker2** [vdemichev/DiaNN#1050] [General]
Dear Vadim!
First of all, thank you so much for your amazing work on DIA-NN. We are now regularly using it for expression proteomics with a lot of success.
One of the things my group is really interested in is residue-specific proteomics with the isoTOP-ABPP platform pioneered by Ben Cravatt (https://doi.org/10.1038/nature18002). Here, we use non-isobaric tags that label e.g. cysteines for quantification. Normally, tools that work for SILAC or dimethyl labeling also work for this, but one challenge is that in peptides with multiple cysteines, only one has the label and the others only carry normal carbamidomethyl. Therefore, if you do not want to use some workarounds, you need to be able to quantify variable modifications (heavy and light) against each other.
We have now used plexDIA in DIA-NN for diaPASEF data for this quantification with OK success. Here, we needed to use the label as fixed modification as otherwise quantification of the labels does not work. In this way, we lose all the peptides with two or more cysteines. Is there a specific reason, why a label can only be defined for a fixed modification in DIA-NN?
Best regards
Stephan

**Comment by vdemichev** [comment]
Hi Stephan,

Interesting use case :) I think it should work fine, if I understood the problem correctly, below are some thoughts.

* --fixed-mod in DIA-NN means somewhat different things depending on the context. For example, when generating a predicted library from FASTA, it means adding a fixed modification to each residue - like expected. When using --channels, it is required to declare a fixed mod for two purposes: first, to let DIA-NN know what this is, and second to enable --lib-fixed-mod (applies fixed mod to any library residue), if the library itself is unlabelled. That is, --fixed-mod and --lib-fixed-mod, in combination, have the same effect on a library, as --fixed-mod alone has on a FASTA.
* Now, if the library is already labelled (in silico) with the base chemical modification in question, then you don't need --lib-fixed-mod, just --fixed-mod alone when analysing with --channels. And --fixed-mod here just declares the meaning of modification, and does not alter the library in any way.
* So how I would do this in this situation:
1. Generate an in silico library from FASTA, specifying both cysteine modifications as variable
2. Analyse this as you would Scenario 4 here https://github.com/vdemichev/DiaNN?tab=readme-ov-file#multiplexing-using-plexdia, that is all as usual, declaring as --fixed-mod (although it was --var-mod when you generated the lib), but omitting --lib-fixed-mod.

Hope this helps! Please let me know if all works here (it should :)).

Best,
Vadim

**Reply by StephanHacker2** [reply]
Hi Vadim!
Thanks so much for the super fast and clear reply.
That is great to know. I will check this out, but it indeed sounds like this could work.
Best regards
Stephan

**Comment by StephanHacker2** [comment]
Hi Vadim!
This worked! Thanks so much for the quick help.
One other thing. Reading the documentation, it sounds to me like for DIA you do not necessarily want to have all possible peptides in the library and that it might help to have only the peptides you are interested in in there. Would it in this case make sense, if you are only interested in the cysteines modified with the tags, to also only include those peptides with a modified cysteine in the speclib?
Best regards
Stephan

**Reply by vdemichev** [reply]
Hi Stephan,

Not exactly, ideally you want everything detectable in the library. So if a particular 'category' of peptides does produce a meaningful number of IDs, makes sense to include this category.

Best,
Vadim

**Comment by StephanHacker2** [comment]
Hi Vadim!
I had one additional question regarding this discussion. I found the following in the readme regarding PTMs:

"Distinguishing between peptidoforms bearing different sets of modifications is a non-trivial problem in DIA: without special peptidoform scoring the effective peptidoform FDR can be in the range 5-10% for library-free analyses. DIA-NN implements a statistical target-decoy approach for peptidoform scoring, which is enabled by the Peptidoforms option (Algorithm pane) and is also activated automatically whenever a variable modification is declared, via the GUI settings or the --var-mod command. The resulting peptidoform q-values reflect DIA-NN's confidence in the correctness of the set of modifications reported for the peptide as well as the correctness of the amino acid sequence identified."

Now I am wondering a bit for the strategy that we discussed above (having the modification as variable in the library generation and then as fixed in the search). In this case, I can probably not do it like this as this only works for variable modifications, right? The parquet files do still have Lib.Peptidoform.Q.Value, Peptidoform.Q.Value and Global.Peptidoform.Q.Value columns, but as there are no variable modifications, is there any benefit to filter for any of these? Would this overall be a concern in the described experiment, where most peptides are supposed to have this modification and we have the channels to compare to each other (and kind of cross-validate heavy and light identification)?

Best regards
Stephan

**Reply by vdemichev** [reply]
Hi Stephan,

You can actually declare a modification as both fixed and variable during search (but not during library generation).
However in this case you would only need it if you want to localise the cysteine modifications. If you don't, can just enable peptidoform scoring, without declaring with --var-mod. But I guess you are right, filtering by channel q-values can make peptidoform scoring somewhat redundant in this case, i.e. most likely OK not to use.

Best,
Vadim

**Reply by StephanHacker2** [reply]
Dear Vadim!
Thanks so much. That is very helpful. I will play around with this a bit.
Indeed localization within the peptide is not super critical. I guess the main thing is to be sure about the peptide sequence and the kind of modification. So I will try the two following things:
- during the search, adding it as --var-mod as well and enable peptidoforms
- during the search, only having it as --fixed-mod and still enable peptidoforms
I do filter by q.vlaues and global.q.values in both channels and in the end the ion will only be retained, if it passes the filter in both channels. So, I would think that this also gives additional confidence in the modification. Nevertheless, an additional peptidoform.q.value/global.peptidoform.q.value filter, if possible, might be on the safe side, I guess.
Best regards
Stephan

**Reply by vdemichev** [reply]
Hi Stephan,

For channel-confidence, need to also use speicifically the Channel.Q.Value (precursor-level) or PG.Q.Value (protein-level).

Best,
Vadim

**Reply by StephanHacker2** [reply]
Hi Vadim!
Still working on implementing this.
The idea for my data analysis is that I start from the report.parquet file, do FDR filters first, then I combine the same ion for the two channels, filter this for MS1.Profile.Corr (>0.7), calculate a ratio for each ion and then I combine all ions with the same modified cysteine. This is essentially, what we want to report; a log2 ratio of heavy to light intensity for each cysteine.
With everything else, I am rather sure that it is OK, but the FDR filtering for the Q values is something I am unsure about. I currently filter on Q.vlaue, PG.Q.value, Global.Q.value and Global.PG.Q.value all for maximum 0.01. Nevertheless, there is also Lib.Q.value, Lib.PG.Qvalue, Peptidoform.Q.value, Global.Peptidoform.Q.value, Lib.Peptidoform.Q.value, Translated.Q.value, Channel.Q.value and some more, which I think are not relevant. Can there be a general guidance, for which of these the data should be filtered in my case? Is 0.01 reasonable on all of them? I am a bit confused to be honest.
Best regards
Stephan

**Reply by StephanHacker2** [reply]
Hi Vadim!
After a certain break, I am getting started on this again. If at all possible, it would be great to get some guidance on the comment one above. All other things, I feel quite confident that I can benchmark myself, but for the FDR, I find it hard to judge, what is the right way.
Best regards
Stephan

**Reply by vdemichev** [reply]
Hi Stephan,

I assume you are analysing with MBR and you'd like channel-specific confidence (always in case you use --channels). This means filters based on:
- Q.Value
- Lib.Q.Value
- Lib.PG.Q.Value
- Channel.Q.Value
- PG.Q.Value (will be channel-specific)
- Lib.Peptidoform.Q.Value and Peptidoform.Q.Value if you'd like peptidoform confidence

Peptidoform and channel-specific filters often OK to be set to values up to, say, 0.05. 

Best,
Vadim

**Reply by StephanHacker2** [reply]
Hi Vadim!

Thanks so much.

Yes, I do use --channels and we do use MBR. So, I guess these are a given to be filtered for:
Q.Value
Lib.Q.Value
Lib.PG.Q.Value
Channel.Q.Value
PG.Q.Value (will be channel-specific)

(So far, we used Global.Q.value and Global.PG.Q.value, but I understand that those should be replaced by Lib.Q.value and Lib.PG.Q.value)

For this one, I am not entirely sure:
Lib.Peptidoform.Q.Value and Peptidoform.Q.Value if you'd like peptidoform confidence

Does this mean, I should only use, it if it is really important that the localization of the modification is also correct with that FDR?

Best regards
Stephan

**Reply by vdemichev** [reply]
>  it if it is really important that the localization of the modification is also correct with that FDR?

Not localisation (for localisation, special localisation scores for modifications declared with --var-mod), but rather confirming that the modification is indeed present on the peptide.

**Reply by StephanHacker2** [reply]
Hi Vadim!
Thanks so much. That makes sense. In that case, it is very likely relevant for us to keep that filter in.
Best regards
Stephan

**Comment by StephanHacker2** [comment]
Hi Vadim!

Thanks so much for all of your help. I am now confident in the filters that we use and up to DIA-NN version 1.9.1 we are getting very interesting data.

I have now tried to also go to version 1.9.2 and tested it with our setup (as a reminder, kind of a plexDIA setup, where the modification is on cysteine).
We use the following for library generation:
‐‐var‐mod IA_isoDTB, 519.2918, C
--original-mods
--min-fr 16
And the following in the search:
--fixed-mod IA_isoDTB, 519.2918, C
--channels IA_isoDTB,0,C,0; IA_isoDTB,6,C,6.0075
--peak-translation
--original-mods
--report-lib-info
--ms1-subtract 2
--channel-run-norm

Now it looks like that in the newest version of DIA-NN (1.9.2) the MS2 level quant seems to work (Precursor.Quant and Precursor.Normalized values as well as Quantity.Quality are reasonable and in agreement with the expectation, but the MS1 level quant (Ms1.Area, MS1.Normalized, Ms1.Apex.Area as well as MS1.profile.correlation) seem to be off. Sadly, we use filters on MS1.Profile.Correlation on our data and use MS1.Area as basis for our quantification. It seems to me like the Channel 0 quantifies OK, but the Channel 6 does not find the right peak (very low intensity and very low MS1.Profile.Correlation for the vast majority of peaks).

We are benchmarking with 1:1 and 4:1 mixed data and the older versions (1.9.1 and 1.9) give expected results with the same workflow, filters and downstream processing. So, it seems to be specific to 1.9.2.

Not sure, if you can help with this or if you would need any additional input, but I would be very happy about any help.

Best regards
Stephan

**Comment by vdemichev** [comment]
Hi Stephan,

I will check if maybe something wrong with --ms1-subtract in the new version. For now would suggest to see if not including this option solves the issue.

Best,
Vadim

**Reply by StephanHacker2** [reply]
Hi Vadim!
Disabling this option worked. If it is disabled, I get comparable results in DIA-NN 1.9.1 and 1.9.2. Nevertheless, having it enabled seems to help in 1.9.1.
Is there an explanation somewhere, what this command does exactly and what the parameter that it is set to (here 2) describes?
Best regards
Stephan

**Reply by vdemichev** [reply]
Hi Stephan,

It subtracts -1 Da signal from the monoisotopic signal, i.e. if the monoisotopic signal is primarily an interference from a channel with low mass, then this option makes it 0. I did test it in 1.9.2 after you reported an issue, and on the dataset I used so far it appeared that it worked OK, I will investigate further.

Best,
Vadim
